test/spl/degree_elevation_test.cc

```diff
@@ -0,0 +1,516 @@
+              original_->GetKnotVector(0)->GetNumberOfKnots()
+                  + original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+  ASSERT_THAT(after_elevation_->GetKnotVector(1)->GetNumberOfKnots(), original_->GetKnotVector(1)->GetNumberOfKnots());
+}
+
+TEST_F(A2DBSplineForDegreeElevation, HasMoreControlPoints) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetPointsPerDirection()[0],
+              original_->GetPointsPerDirection()[0] + original_->GetKnotVector(0)->GetNumberOfDifferentKnots() - 1);
+  ASSERT_THAT(after_elevation_->GetPointsPerDirection()[1], original_->GetPointsPerDirection()[1]);
+}
+
+TEST_F(A2DBSplineForDegreeElevation, DoesNotChangeGeometricallyAfterDegreeElevation) {  // NOLINT
+  ASSERT_THAT(after_elevation_->AreGeometricallyEqual(*original_), true);
+}
+
+class Random2DBSplineForDegreeElevationInDirection0 : public Test {  // NOLINT
+ public:
+  Random2DBSplineForDegreeElevationInDirection0() {
+    std::array<ParamCoord, 2> limits = {ParamCoord{0.0}, ParamCoord{1.0}};
+    spl::RandomBSplineGenerator<2> spline_generator(limits, 2, 3);
+    spl::BSpline<2> b_spline(spline_generator);
+    original_ = std::make_shared<spl::BSpline<2>>(b_spline);
+
+    spl::BSpline<2> elevation_spline(b_spline);
+    elevation_spline.ElevateDegree(0);
+    after_elevation_ = std::make_shared<spl::BSpline<2>>(elevation_spline);
+  }
+
+ protected:
+  std::shared_ptr<spl::BSpline<2>> original_;
+  std::shared_ptr<spl::BSpline<2>> after_elevation_;
+};
+
+TEST_F(Random2DBSplineForDegreeElevationInDirection0, HasELevatedDegree) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetDegree(0).get(), original_->GetDegree(0).get() + 1);
+  ASSERT_THAT(after_elevation_->GetDegree(1).get(), original_->GetDegree(1).get());
+}
+
+TEST_F(Random2DBSplineForDegreeElevationInDirection0, HasMoreKnots) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetKnotVector(0)->GetNumberOfDifferentKnots(),
+              original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+  ASSERT_THAT(after_elevation_->GetKnotVector(0)->GetNumberOfKnots(),
+              original_->GetKnotVector(0)->GetNumberOfKnots()
+                  + original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+  ASSERT_THAT(after_elevation_->GetKnotVector(1)->GetNumberOfKnots(), original_->GetKnotVector(1)->GetNumberOfKnots());
+}
+
+TEST_F(Random2DBSplineForDegreeElevationInDirection0, HasMoreControlPoints) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetPointsPerDirection()[0],
+              original_->GetPointsPerDirection()[0] + original_->GetKnotVector(0)->GetNumberOfDifferentKnots() - 1);
+  ASSERT_THAT(after_elevation_->GetPointsPerDirection()[1], original_->GetPointsPerDirection()[1]);
+}
+
+TEST_F(Random2DBSplineForDegreeElevationInDirection0, DoesNotChangeGeometricallyAfterDegreeElevation) {  // NOLINT
+  ASSERT_THAT(after_elevation_->AreGeometricallyEqual(*original_), true);
+}
+
+class Random2DBSplineForDegreeElevationInDirection1 : public Test {  // NOLINT
```

Comment:
Does this really test something new? Only keep mixed portion, i.e., after degree elevating 2x for 0th and 1x for 1st direction.